crypto/fipsmodule/rsa/rsa.c

```diff
@@ -916,3 +919,16 @@ int RSA_flags(const RSA *rsa) { return rsa->flags; }
 int RSA_blinding_on(RSA *rsa, BN_CTX *ctx) {
   return 1;
 }
+
+int RSA_pkey_ctx_ctrl(EVP_PKEY_CTX *ctx, int optype, int cmd, int p1, void *p2)
+{
+  /* If key type not RSA or RSA-PSS return error */
+  if (ctx != NULL && ctx->pmeth != NULL 
+      && ctx->pmeth->pkey_id != EVP_PKEY_RSA
+      && ctx->pmeth->pkey_id != EVP_PKEY_RSA_PSS) {
+    // TODO(shang): investigate if a new error reason should be defined to replace EVP_R_OPERATION_NOT_SUPPORTED_FOR_THIS_KEYTYPE.
```

Comment:
I think this can be related to my comment about why this function is in `rsa.c`. 

`EVP_R_OPERATION_NOT_SUPPORTED_FOR_THIS_KEYTYPE` is an EVP level error code.

I would like to hear about your design choices.

Reply:
I will define new error type `RSA_R_OPERATION_NOT_SUPPORTED_FOR_THIS_KEYTYPE`.